core/io/pwrite_spec.rb

```diff
@@ -28,16 +28,42 @@
       @file.pread(6, 0).should == "foobar"
     end
 
+    it "calls #to_s on the object to be written" do
+      data = mock("to_s")
+      data.should_receive(:to_s).and_return("foo")
+      @file.pwrite(data, 0)
+      @file.pread(3, 0).should == "foo"
+    end
+
+    it "calls #to_int on the offset" do
+      offset = mock("to_int")
+      offset.should_receive(:to_int).and_return(2)
+      @file.pwrite("foo", offset)
+      @file.pread(3, 2).should == "foo"
+    end
+
     it "raises IOError when file is not open in write mode" do
       File.open(@fname, "r") do |file|
-        -> { file.pwrite("foo", 1) }.should raise_error(IOError)
+        -> { file.pwrite("foo", 1) }.should raise_error(IOError, "not opened for writing")
       end
     end
 
     it "raises IOError when file is closed" do
       file = File.open(@fname, "w+")
       file.close
-      -> { file.pwrite("foo", 1) }.should raise_error(IOError)
+      -> { file.pwrite("foo", 1) }.should raise_error(IOError, "closed stream")
+    end
+
+    it "raises a NoMethodError if #to_s cannot be called on the object to be written" do
```

Comment:
```suggestion
    it "raises a NoMethodError if object doesn't respond to #to_s" do
```

Reply:
Updated. I changed "doesn't" to "does not" to make it a bit more formal.